api/server/services/auth.service.js

```diff
@@ -62,61 +57,58 @@ const registerUser = async (user) => {
     const existingUser = await User.findOne({ email });
 
     if (existingUser) {
-      log({
-        title: 'Register User - Email in use',
-        parameters: [
-          { name: 'Request params:', value: user },
-          { name: 'Existing user:', value: existingUser }
-        ]
-      });
-      response = { status: 422, message: 'Email is in use' };
-      return response;
+      console.info(
+        'Register User - Email in use',
+        { name: 'Request params:', value: user },
+        { name: 'Existing user:', value: existingUser }
+      );
+      
+      // Sleep for 1 second
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+
+      // TODO: We should change the process to always email and be generic is signup works or fails (user enum)
+      return { status: 500, message: 'Something went wrong' };
     }
 
     //determine if this is the first registered user (not counting anonymous_user)
     const isFirstRegisteredUser = (await User.countDocuments({})) === 0;
 
-    try {
-      const newUser = await new User({
-        provider: 'local',
-        email,
-        password,
-        username,
-        name,
-        avatar: null,
-        role: isFirstRegisteredUser ? 'ADMIN' : 'USER'
-      });
-
-      // todo: implement refresh token
-      // const refreshToken = newUser.generateRefreshToken();
-      // newUser.refreshToken.push({ refreshToken });
-      bcrypt.genSalt(10, (err, salt) => {
-        bcrypt.hash(newUser.password, salt, (errh, hash) => {
-          if (err) {
-            console.log(err);
-          }
-          // set pasword to hash
-          newUser.password = hash;
-          newUser.save();
-        });
-      });
-      console.log('newUser', newUser);
-      if (isFirstRegisteredUser) {
-        migrateDataToFirstUser(newUser);
-        // console.log(migrate);
-      }
-      response = { status: 200, user: newUser };
-      return response;
-    } catch (err) {
-      response = { status: 500, message: err.message };
-      return response;
+    const newUser = await new User({
+      provider: 'local',
+      email,
+      password,
+      username,
+      name,
+      avatar: null,
+      role: isFirstRegisteredUser ? 'ADMIN' : 'USER'
+    });
+
+    // todo: implement refresh token
+
+    // Perform synchronously to prevent a race condition
+    const salt = await bcrypt.genSalt(10);
```

Comment:
just fyi - i have a PR that remove bcrypt and replaces with bcryptjs because bcrypt has a bunch of known issues, so you'll get a couple of merge conflicts but they should be very easy to resolve.